### PR TITLE
Added Swagger Annotations and a Documentation Resource

### LINKED ISSUE #834: Add Swagger Annotations and a Documentation Resource
Annotate the Server Resources and add a Resource that serves the static Swagger json/yaml.

### PR DESCRIPTION
**Pull request by JAmmermann-DLR**
- Added documentation via annotating the bean classes
- Updated the server pom to now create seperate documentation for the Model and Management API
- Generated the swagger documentation into doc-gen directory
- Update docs feature:
  - Added dependency to server
  - Added section for model inheritance
  - Added generated documentation
- Added a DocumetationResource that serves the swagger.json/.yaml via the Rest API

So we should be able to deploy a complete server documentation after this PR :)

There are less changes here than git says, because I added a lot of generated files and it doesn't show the right changes for some bean files :)

Closes #834

### CODE REVIEW

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
Great work, only super minor comments!

File: de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/DocumentationResource.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		
+    	String content = FILE_NOT_FOUND;
+    	String realmResourceName = "doc-gen" + File.separator + directory + File.separator + fileName;
+    	
+		try {
+			InputStream inputStream  = FileLocator.openStream(Activator.getDefault().getBundle(), 
+					new org.eclipse.core.runtime.Path(realmResourceName), false);
+			content = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (JAmmermann-DLR) [reply de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/resources/DocumentationResource.java:72]: No I will change it

File: de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/servlet/RepoManagementServlet.java [hunk 2]
```diff
@@ -31,8 +33,22 @@ protected Servlet onCreateServlet() {
...
 	private static class RepoManagementRestApplication extends ResourceConfig {
+		/**
+		 * Registers all relevant Classes: Resources, Filter and Bindings
+		 */
 		private RepoManagementRestApplication() {
 			register(ProjectManagementResource.class);
+
+			// Register documentation resource via binder
+			final DocumentationResource docProvider = new DocumentationResource("management");
+			final AbstractBinder docBinder = new AbstractBinder() {
```
[the remark this answers is not in the document]
Comment (JAmmermann-DLR) [reply de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/servlet/RepoManagementServlet.java:44]: This seems to only work on interfaces and not on abstract classes (Compiler Error: "The target type of this expression must be a functional interface")

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Review by andriikovalov-dlr — APPROVED** [review]

**Pull request merged**